src/App.tsx

```diff
@@ -1,26 +1,257 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+
 import { UserWarning } from './UserWarning';
+import * as service from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoItem } from './components/TodoItem/TodoItem';
 
-const USER_ID = 0;
+enum Filter {
+  ALL = 'ALL',
+  ACTIVE = 'ACTIVE',
+  COMPLETED = 'COMPLETED',
+}
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterTodos, setFilterTodos] = useState<Filter>(Filter.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [todoTitle, setTodoTitle] = useState<string>('');
+  const [errorMessage, setErrorMessage] = useState<string>('');
+
+  useEffect(() => {
+    service
+      .getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'));
+  }, []);
+
+  useEffect(() => {
+    if (errorMessage) {
+      const timer = setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+
+      return () => clearTimeout(timer);
+    }
+
+    return undefined;
+  }, [errorMessage]);
+
+  const filteredTodos = todos.filter(todo => {
+    if (filterTodos === 'ACTIVE') {
```

Comment:
```suggestion
    if (filterTodos === Filter.Active) {
```
you should always use enum values like this